### DISCUSSION TITLE
Configuration required from mutiple network connectivity

### DISCUSSION THREAD

**Discussion by Sumitsrjhs** [strongswan/strongswan#2401] [Q&A]
On-Prem ------------------------- AWS ------------------------------ GCP
(192.168.x.x)-------------------- (172.10.x.x)----------------- (10.10.x.x)
                                                

Hi Friends

I have configured Strongswan on ec2 instance on AWS. I can access GCP Servers(10.10.x.x) from AWS (172.10.x.x) as well as from On-Prem (192.168.x.x).

I can access On-Prem (192.168.x.x) from AWS (172.10.x.x). Also AWS (172.10.x.x) from GCP (10.10.x.x).

But I am not able to access On-prem(192.168.x.x) from GCP (10.10.x.x)

I have also configured masquerading

iptables -t nat -A POSTROUTING -o ens5 -j MASQUERADE 

Please suggest

Thanks

**Comment by strongX509** [comment]
If you have an IPsec tunnel between A and B as well as a second IPsec tunnel between C and B then the IPsec policies do not allow tunneling packets from A to C (This is how the IPsec standard works). You have to extend the local and remote IP traffic selectors on each host accordingly so that packet forwarding from A to C via B becomes possible.

**Reply by Sumitsrjhs** [reply]
Hi Strong

Thanks for the clarification. If I assume on prem as A, AWS as B and GCP as C. my traffic is forwarding without any hurdle from A to C using masquerading. But in reverse it's not forwarding from C to A. However I can ping from C to B and B to A.

Please suggest

**Reply by jxcang** [reply]
I would ask you to clarify what is ens5 interface. I need to guess that is the interface on AWS connecting to GRE.
Regardless, It sounds like you need to convert a policy-based VPN to a route-based VPN.

There are multiple ways to achieve this.
(1) Stick with the policy-based IPsec, which is the simplest and stick to what you have. Add multiple subnets to the remote_ts and the local_ts.
(2) switch to route-based IPsec, and configure IP-IP or GRE tunnels inside
(2) switch to route-based IPsec, and use VTI packet marking or XFRM interfaces

**Reply by Sumitsrjhs** [reply]
Hi jxcang

Thanks for the solution you provided. I am going to start working on your suggestions.
Yes, ens5 is the network interface for EC2 instance on AWS.
I will let you know if I need any more help.

Thanks.

**Reply by Sumitsrjhs** [reply]
Hi jxlang

I have reviewed my setup and realized that i am using ipsec.conf instead of swanctl.conf.
My ipsec.conf is 
	left=172.10.x.x, 10.10.x.x/x
        leftsourceip=x.x.x.x
        leftsubnet=172.10.x.x/x, 10.10.x.x/x
        leftid=x.x.x.x
        right=x.x.x.x
        rightid=x.x.x.x
         esp=aes256-sha1-modp2048
        ike=aes256-sha1-modp2048
        authby=psk
        keyexchange=ikev2
        rightsubnet=192.168.x.x/x
        auto=start
        ikelifetime=1h
        lifetime=8h
        dpddelay=30
        dpdtimeout=120
        dpdaction=restart
        leftfirewall=no
        rightfirewall=no

I have tried to change leftsubnet and rightsubnet to 0.0.0.0/0 but its still not working as expected.
Please suggest
